Library/Homebrew/sorbet/tapioca/compilers/args.rb

```diff
@@ -0,0 +1,68 @@
+# typed: strict
+# frozen_string_literal: true
+
+require_relative "../../../global"
+
+# require all the commands
+["cmd", "dev-cmd"].each do |dir|
+  Dir[File.join(__dir__, "../../../#{dir}", "*.rb")].each { require(_1) }
+end
+
+module Tapioca
+  module Compilers
+    class Args < Tapioca::Dsl::Compiler
+      # This is ugly, but we're moving to a new interface that will use a consistent DSL
+      # These are cmd/dev-cmd methods that end in `_args` but are not parsers
+      NON_PARSER_ARGS_METHODS = T.let([
+        :formulae_all_installs_from_args,
+        :reproducible_gnutar_args,
+        :tar_args,
+      ].freeze, T::Array[Symbol])
+
+      # FIXME: Enable cop again when https://github.com/sorbet/sorbet/issues/3532 is fixed.
+      # rubocop:disable Style/MutableConstant
+      ConstantType = type_member { { fixed: T.class_of(Homebrew::CLI::Args) } }
+      # rubocop:enable Style/MutableConstant
+
+      sig { override.returns(T::Enumerable[T.class_of(Homebrew::CLI::Args)]) }
+      def self.gather_constants = [Homebrew::CLI::Args]
+
+      sig { override.void }
+      def decorate
+        root.create_path(Homebrew::CLI::Args) do |klass|
+          Homebrew.methods(false).select { _1.end_with?("_args") }.each do |args_method_name|
+            next if NON_PARSER_ARGS_METHODS.include?(args_method_name)
+
+            parser = Homebrew.method(args_method_name).call
+            comma_array_methods = comma_arrays(parser)
+            args = parser.instance_variable_get(:@args)
+            args.instance_variable_get(:@table).each do |method_name, value|
```

Comment:
We're pretty deep in the internals of the parser at this point (accessing the ivar of an ivar), it might be a good idea to write tests and/or update CLI::Parser to more directly expose its method table.

Reply:
Those sound like good ideas. If we expose `:@table`, we should make sure that it's read only.

Reply:
Ok, I broke up the logic in order to test various aspects of this compiler. (I decided not make any changes to CLI::Parser). PTAL.